### DISCUSSION TITLE
Validation rule names such as Alpha and AlphaNum are not intuitive in some languages

### DISCUSSION THREAD

**Discussion by fuwasegu** [laravel/framework#45660] [General] [answered]
Regarding the Alpha Validation rules, the Laravel documentation states
> The field under validation must be entirely alphabetic characters.

And the actual implementation logic is as follows
```php
public function validateAlpha($attribute, $value)
{
    return is_string($value) && preg_match('/^[\pL\pM]+$/u', $value);
}
```

According to the distribution map in [this Wikipedia(List of Unicode characters)](https://en.wikipedia.org/wiki/List_of_Unicode_characters) , the characters in `\p{L}` are all alphabetic in the Unicode system, so the validation logic itself is correct.
![image](https://user-images.githubusercontent.com/52437973/212629000-56285482-0468-4277-a437-3bb154d1cdec.png)

However, the common "alphabet" in Japan refers to the 26 letters used in English, from A-Z.

As a reference, the authoritative Japanese dictionary "Kojien" defines the alphabet as follows
1. a generic term for a writing system in which letters represent phonemes. Latin alphabet, Hebrew alphabet, Arabic alphabet, etc. Based on the first two letters of the Greek alphabet, alpha (alpha) and beta (beta), which were called together.
2. the 26 letters used in the Roman alphabet, especially in English.

As far as I know, most Japanese people take "alphabet" in the second sense.

In short, when Japanese hear "The field under validation must be entirely alphabetic characters.", they imagine a rule equivalent to "a-zA-Z".

I submitted a PR like https://github.com/laravel/framework/pull/45609, but it was closed.

Based on the possibility that many Japanese may use this validation rule with a misunderstanding, we believe that we should at least warn them in the document.
If possible, please consider modifying the validation logic or adding a new validation rule like the PR I submitted.

Similar issues and PRs have been filed in the past, but there seems to be no resolution.
https://github.com/laravel/framework/issues/43624
https://github.com/laravel/framework/issues/42831
https://github.com/laravel/framework/pull/43683

**Comment by siketyan** [comment]
I can not imagine how the people thinks about that who speaks English as the mother tongue; Do the people really think that the "Alphabetic characters" means the full set of alphabetical characters including Kana and so on defined in the Unicode?

**Comment by mpyw** [comment]
It should be considered that there are indispensable divergence between the strict Unicode definition and the nuances that are imagined in everyday life. Laravel is not a tool for academics, but a Web framework used more broadly by the masses.

**Comment by mpyw** [comment]
@taylorotwell @driesvints @nunomaduro @doshu @hmingv @joelclermont I would like to mention for the maintainers and related issue participants, thank you for your time. Any opinions?

**Comment by hmingv** [comment]
My purpose is to let customers only enter a-zA-Z.

"alphabetic" I will translate into "字母(letters)", in my concept it is the English alphabet, of course, I can give up the understanding of Chinese, it is "alphabetic".

Just, if it covers a really wide range, why can't we use "words" to describe it.

https://en.wikipedia.org/wiki/Alphabet

**Reply by hmingv** [reply]
Also, I tried https://forge.laravel.com/auth/register.

I can use Chinese as password.

**Comment by fuwasegu** [comment]
I have read RFC2234 as a supplement to this.
RFC2234 defines ABNF, a language for describing formal systems such as communication protocols.
Section 6.1 of this RFC defines the core rules for writing ABNF.
According to this, "ALPHA" is defined as representing "A-Z / a-z".

https://datatracker.ietf.org/doc/html/rfc2234#section-6.1
![image](https://user-images.githubusercontent.com/52437973/213614832-4c4321e6-e816-4740-ad11-c630b3488d91.png)


In my argument above, I was mainly talking about "how alphabets are regarded in a language area where multi-byte characters are commonly used, such as Japan", but looking at this RFC, it seems that ALPHA generally refers to the 26 characters of A-Z (or its upper case and lower case combined, 52 characters) even in the computer world.

**Comment by fuwasegu** [comment, marked as answer]
https://github.com/laravel/framework/pull/45769
This PR was merged then we can use `alpha:ascii` rule with the option for strict validation!
